cloud.foundry.cli/src/main/java/cloud/foundry/cli/crosscutting/mapping/CfArgumentsCreator.java

```diff
@@ -90,7 +111,49 @@
                 args.add(prop.getProperty(key));
 
                 log.info("Extended CommandLine Argument with the Option: " + key +
-                        " and value " + "'" + prop.getProperty(key) + "'");
+                    " and value " + "'" + prop.getProperty(key) + "'");
+            });
+        } catch (IOException ex) {
+            log.error(ex.getMessage());
+            System.exit(1);
+        }
+
+        return args.toArray(new String[0]);
+    }
+
+    /**
+     * Extends the commandLine with the missing options by reading the given yaml file.
+     *
+     * @param missingOptions missing Options like -s, -a, -o
+     * @param args           Commandline arguments
+     * @return Commandline arguments
+     */
+    private static String[] extendForDiffAndApplyCommand(List<String> missingOptions, LinkedList<String> args) {
+        try {
+            String yamlPath = args.get(args.indexOf("-y") + 1);
+            ConfigBean configBean = YamlMapper.loadBeanFromFile(yamlPath, ConfigBean.class);
+            TargetBean targetBean = configBean.getTarget();
+
+            missingOptions.forEach(key -> {
+                args.add(key);
+                String value = "";
+                switch (key) {
+                case "-a":
+                    value = targetBean.getEndpoint();
+                    break;
+                case "-o":
+                    value = targetBean.getOrg();
+                    break;
+                case "-s":
+                    value = targetBean.getSpace();
+                    break;
+                default:
+                    break;
+                }
+                
+                args.add(value);
+                log.info("Extended CommandLine Argument with the Option: " + key +
+                    " and value " + "'" + value + "'");
             });
         } catch (IOException ex) {
```

Comment:
This catch has to catch all exceptions that might be thrown due to the call to `YamlMapper.loadBeanFromFile(...)`. Take a look at its documentation and add exception handling for each possbile type of exception